presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -641,6 +717,27 @@ public void testPushdownSubfieldsAssorted()
                         "a", toSubfields("a[1]"),
                         "d", toSubfields("d.d3[5]")));
 
+        //Subfield pruning should pass-through arbitrary() function
+        assertPushdownSubfields("SELECT id, " +
+                        "arbitrary(x.a), arbitrary(x).a, arbitrary(x).d.d1, arbitrary(x.d).d1, arbitrary(x.d.d2), " +
+                        "arbitrary(y[1]).a, arbitrary(y[1]).d.d1, arbitrary(y[2]).d.d1, arbitrary(y[3].d.d1), " +
+                        "arbitrary(z).c, " +
+                        "arbitrary(w[1][2]).e.e1, arbitrary(w[2][3].e.e2) " +
+                        "FROM test_pushdown_subfields " +
+                        "GROUP BY 1", "test_pushdown_subfields",
+                ImmutableMap.of("x", toSubfields("x.a", "x.d.d1", "x.d.d2"),
+                        "y", toSubfields("y[1].a", "y[1].d.d1", "y[2].d.d1", "y[3].d.d1"),
+                        "z", toSubfields("z.c"),
+                        "w", toSubfields("w[1][2].e.e1", "w[2][3].e.e2")));
+
+        //Subfield pruning should not pass-through other aggregate functions e.g. min() function
+        assertPushdownSubfields("SELECT id, min(x.d).d1, min(x.d.d2), min(z).c, min(z.b), min(y[1]).a, min(y[1]).d.d1, min(y[2].d.d1), min(w[1][2]).e.e1, min(w[2][3].e.e2) " +
```

Comment:
nit: perhaps, put each `min` on a separate line for readability

Reply:
Done